### ISSUE TITLE
Popup - create integration with background and actions

### ISSUE THREAD

**Issue by wirednkod** [paritytech/substrate-connect#95]
Tasks:
- [ ] Create listener for events emitted on open/new tab creation
- [ ] Create listener for events emitted on change of tab contents (add/remove uApp, network/uApp status)
- [ ] Create listener for events emitted on destroy/close tab
- [ ] Create API call integration for the state of uApps/tabs

**Comment by wirednkod** [comment]
#93

**Comment by wirednkod** [comment]
Parent issue: #93

**Comment by wirednkod** [comment]
@raoulmillais @goldsteinsveta 

### API used by developer on the page

```javascript
import { ApiPromise } from '@polkadot/api';
import { UApp } from './';

const burnrApp = new UApp('burnr-wallet'); /// This is the name of the UApp
// check for extension - this is the name of the network - it also detects the page URL and the tabId
const provider = burnrApp.detect('westend'); 
await provider.connect();
const api = await ApiPromise.create({ provider });

// interact with polkadot js API

// ...

provider.disconnect()

const browserDemoApp = new UApp('smoldot-browser-demo'); /// This is the name of the UApp
// check for extension - this is the name of the network - it also detects the page URL and the tabId
const provider2 = uapp.detect('westend'); 
await provider2.connect();
const browserDemoAppApi = await ApiPromise.create({ provider });

```




#### Shared data structure between background and UI

How to get the shared datastructure from react:

- UI API

```javascript
// optional tabId
const listener = chrome.backgroundPage().getStateListener(); 
const state = chrome.backgroundPage().getState(1234 /* tabId */); 

// happens the first time we see a UApp connect from a tab
stateListener.on('tab:created', (tabId) => {});
stateListener.on('tab:changed', (tabId) => {});
stateListener.on('tab:destroyed', (tabId) => {});
```


- State

```javascript
{
  tabId: 0,
  url: 'my-awesome-uapp.com/index.html',
  uApps: [{
      networks: [{'westend', status: 'connected'}],
      name: '',  // unique name supplied by developer
      enabled: true,
    },
    {
      networks: [{ 'westend', status: 'connected'}],
      name: '',
      enabled: false
    },
    {}
   ]	
},
{
  tabId: 1,
  url: 'my-awesome-uapp.com/index.html',
  uApps: [{
      networks: [{ 'westend', status: 'connected'}],
      name: '',
      enabled: true
    },
    {
      networks: [{ 'westend', status: 'connected'}],
      name: '',
      enabled: false
    },
    {}
   ]	
}
```

### Other unrelated tasks - separate issue

Garbage collect smoldot client when there are no more uapps connected to a network

**Comment by wirednkod** [comment]
Dependency on #94 and #126 

@raoulmillais Are any of the tasks mentioned [here](https://github.com/paritytech/substrate-connect/issues/95#issue-804632424) resolved from #126 ?
